Add more details about Firebase hosting (#8804)
flutter/website: src/deployment/web.md

```

{{site.alert.note}}

> The `canvaskit` directory and its contents are only present when the
> CanvasKit renderer is selected—not when the HTML renderer is selected.

Reviewer: ```suggestion
  The `canvaskit` directory and its contents are only present when the
  CanvasKit renderer is selected—not when the HTML renderer is selected.
```

> The `canvaskit` directory and its contents are only present when the
> CanvasKit renderer is selected—not when the HTML renderer is selected.

Reviewer 2: **issue:** Write as shorter sentences per [GSG](https://developers.google.com/style/accessibility#ease-of-reading).  Write in active voice per [GSG](https://developers.google.com/style/voice).
```suggestion
When you select the CanvaKit renderer, the `canvaskit` directory
and its contents appear. When you select the HTML renderer,
this directory doesn't appear.
```

(given the python SimpleHTTPServer example)
to view the release version of your app.

## Deploying to the web

When you are ready to deploy your app,
upload the release bundle
to Firebase, the cloud, or a similar service.
Here are a few possibilities, but there are
many others:

* [Firebase Hosting][]
* [GitHub Pages][]
* [Google Cloud Hosting][]

## Deploying to Firebase Hosting
You can use the Firebase CLI to build and release your Flutter app with Firebase
Hosting.

### Before you begin
To get started, [install or update][install-firebase-cli] the Firebase CLI:

```
npm install -g firebase-tools
```

### Initialize Firebase

1. Enable the web frameworks preview to the [Firebase framework-aware CLI][]:

    ```
    firebase experiments:enable webframeworks
    ```

2. In an empty directory or an existing Flutter project, run the initialization
command:

    ```
    firebase init hosting
    ```

> 3. Answer yes when asked if you want to use a web framework.

Reviewer: ```suggestion
3. Answer `yes` when asked if you want to use a web framework.
```

> 4. If you are in an empty directory, you will be asked to choose your web
> framework. Choose Flutter Web.

Reviewer: ```suggestion
4. If you're in an empty directory,
    you'll be asked to choose your web framework. Choose `Flutter Web`.
```

> 5. Choose your hosting source directory, this could be an existing flutter app.

Reviewer: ```suggestion
5. Choose your hosting source directory; this could be an existing flutter app.
```

6. Select a region to host your files.

7. Choose whether to set up automatic builds and deploys with GitHub.

8. Deploy the app to Firebase Hosting:

>     ```

Reviewer 2: **chore:** Set this to `terminal`.
```suggestion
    ```terminal
```

    firebase deploy
    ```

>     Running this command automatically runs `flutter build web --release` for you,

Reviewer: ```suggestion
    Running this command automatically runs `flutter build web --release`,
```

>     Running this command automatically runs `flutter build web --release` for you,
>     so you don't have to build your app in a separate step.

Reviewer 2: **issue:** 
Write as shorter sentences per [GSG](https://developers.google.com/style/accessibility#ease-of-reading).
```suggestion
    By running this command, Flutter invokes `flutter build web --release`.
    You don't have to build your app in a separate step.
```

> For more information, visit the official [Firebase Hosting][] documentation for
> Flutter on the web.

Reviewer 2: **nit:** Phrase as a call to action.
```suggestion
To learn more, visit the official [Firebase Hosting][] documentation for
Flutter on the web.
```

> For more information, visit the official [Firebase Hosting][] documentation for
> Flutter on the web.

Reviewer 2: **nit:** Recommend a call to action instead.
```suggestion
To learn more, visit the official [Firebase Hosting][] documentation for
Flutter on the web.
```

## Handling images on the web

> The web supports the standard `Image` widget to display images.
> However, because web browsers are built to run untrusted code safely,
> there are certain limitations in what you can do with images compared
> to mobile and desktop platforms.
> 

Reviewer 2: **issues:** Write as shorter sentences per [GSG](https://developers.google.com/style/accessibility#ease-of-reading). Write in active voice per [GSG](https://developers.google.com/style/voice).
```suggestion
The web supports the standard `Image` widget to display images.
By design, web browsers run untrusted code without harming the host computer.
This limits what you can do with images compared to mobile and desktop platforms.

```

For more information, see [Displaying images on the web][].

## Choosing a web renderer

By default, the `flutter build` and `flutter run` commands
use the `auto` choice for the web renderer. This means that
your app runs with the HTML renderer on mobile browsers and

> CanvasKit on desktop browsers. This is our recommended combination
> to optimize for the characteristics of each platform.

Reviewer 2: **nit:** Write in active voice per [GSG](https://developers.google.com/style/voice).
```suggestion
CanvasKit on desktop browsers. We recommend this combination
to optimize for the characteristics of each platform.
```

For more information, see [Web renderers][].

## Minification

Minification is handled for you when you
create a release build.

> A debug build of a web app is not minified and
> tree shaking has not been performed.
> 
> A profile build is not minified and tree shaking
> has been performed.
> 
> A release build is both minified and tree shaking
> has been performed.

Reviewer 2: **suggestion:** Simplify as a table:
```suggestion
| Type of web app build | Code minified? | Tree shaking performed? |
|-----------------------|----------------|-------------------------|
| debug                 | No             | No                      |
| profile               | No             | Yes                     |
| release               | Yes            | Yes                     |
```

[dhttpd]: {{site.pub}}/packages/dhttpd
[Displaying images on the web]: {{site.url}}/platform-integration/web/web-images
[Firebase Hosting]: {{site.firebase}}/docs/hosting/frameworks/flutter
[Firebase framework-aware CLI]: {{site.firebase}}/docs/hosting/frameworks/frameworks-overview

> [Install-firebase-cli]: https://firebase.google.com/docs/cli#install_the_firebase_cli

Reviewer: Case matters for links.

```suggestion
[install-firebase-cli]: https://firebase.google.com/docs/cli#install_the_firebase_cli
```
